packages/mui-joy/src/Textarea/Textarea.tsx

```diff
@@ -120,17 +120,15 @@ const TextareaRoot = styled('div', {
       // variant styles
       ...variantStyle,
       backgroundColor: variantStyle?.backgroundColor ?? theme.vars.palette.background.surface,
-      [`&:hover:not(.${textareaClasses.focused})`]: {
+      '&:hover': {
         ...theme.variants[`${ownerState.variant!}Hover`]?.[ownerState.color!],
-        backgroundColor: null, // it is not common to change background on hover for Input
+        backgroundColor: null, // it is not common to change background on hover for Textarea
         cursor: 'text',
       },
       [`&.${textareaClasses.disabled}`]:
         theme.variants[`${ownerState.variant!}Disabled`]?.[ownerState.color!],
-      [`&.${textareaClasses.focused}`]: {
-        '&:before': {
-          boxShadow: `inset 0 0 0 var(--Textarea-focusedThickness) var(--Textarea-focusedHighlight)`,
-        },
+      '&:focus-within::before': {
```

Comment:
One point of complication, on https://mui.com/x/react-date-pickers/date-range-picker/#basic-usage, we found a use case to set the focus style without having the input actually DOM focused 😅

Reply:
> I have noticed one bug related to the `<input>` not taking the whole space:

pushed a fix!

> when clicking on the input but not over the <input> the focus is only set on mouse up, I think that it should be on mouse down. I did it wrong on the Autocomplete too, e.g. https://mui.com/material-ui/react-autocomplete/#combo-box, not a great UX.

This should be fixed from Base UI's `useInput` hook

> One point of complication, on https://mui.com/x/react-date-pickers/date-range-picker/#basic-usage, we found a use case to set the focus style without having the input actually DOM focused 😅

I oppose the use of JS for this case and I think CSS approach is better. Here is how you toggle the focus ring in Joy UI:

```js
<Input sx={{ '--Input-focused': 1 }} />
```
added to the docs: https://deploy-preview-37385--material-ui.netlify.app/joy-ui/react-input/#triggering-the-focus-ring